receiver/kafkareceiver/factory.go

```diff
@@ -53,18 +53,30 @@ func WithAddUnmarshallers(encodingMarshaller map[string]Unmarshaller) FactoryOpt
 	}
 }
 
+// WithAddLogsUnmarshallers adds logsUnmarshallers.
+func WithAddLogsUnmarshallers(encodingMarshaller map[string]logsUnmarshaller) FactoryOption {
```

Comment:
Names are inconsistent, probably worth updating the previous one to include traces.

Also I would change the API to:

```golang
func WithLogsUnmarshallers(encoding string, unmarshaller logsUnmarshaller) FactoryOption {}
```

Or even consider to add a `name` func on the interface to simplify usage.

Reply:
Actually there is an `Encoding` so we can do:

```golang
func WithLogsUnmarshallers(unmarshallers logsUnmarshaller...) FactoryOption {}
```

Reply:
The method is actually consistent with the traces' method here: https://github.com/open-telemetry/opentelemetry-collector/blob/6458914bbd0d06bc3fa6304197fe736d47b0f694/receiver/kafkareceiver/factory.go#L47-L54

I'd like to keep the parameters consistent with trace's method and there are multiple encodings and return one factory in traces' method. But I'd like to rename the method name to be consistent in a subsequent PR. Thoughts?